packages/stream/get-streams/get-streams.test.js

```diff
@@ -0,0 +1,135 @@
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('TypeError')
+        expect(error.message).to.be.equal(
+          'Failed to process because initialization is not valid. Please provide required initialization argument which is the initialization instance returned by the init() function'
+        )
+      }
+
+      try {
+        await getStreams({ tes: 'blabla' })
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('TypeError')
+        expect(error.message).to.be.equal(
+          'Failed to process because initialization is not valid. Please provide required initialization argument which is the initialization instance returned by the init() function'
+        )
+      }
+    })
+
+    it('should return error response if API key is not valid', async function () {
+      nock(`${Internal.config.api.base_url}`)
+        .get(`/${Internal.config.api.version}/streams/`)
+        .reply(403, { code: 403, message: 'API key is not valid', data: '' })
+
+      try {
+        await getStreams(init({ api_key: 'blabla' }))
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('Error')
+        expect(error.message).to.be.equal(
+          'Failed to get a list of streams because the API Key is not valid. Please provide a valid and active API Key.'
+        )
+      }
+    })
+  })
+
+  describe('Positive test', function () {
+    beforeEach(function () {
+      nock(`${Internal.config.api.base_url}`)
+        .get(`/${Internal.config.api.version}/streams/`)
+        .reply(200, {
+          status: {
+            code: 200,
+            message: 'Success',
+            type: 'success',
+          },
+          data: [
+            {
+              id: 371,
+              name: 'a new stream',
+              slug: '',
+              hls_manifest_path: '',
+              dash_manifest_path: '',
+              description: '',
+              created_by: 6,
+              created_at: '2022-09-05T07:38:09.525329768Z',
+              updated_at: '2022-09-05T07:38:09.525329768Z',
+              quality: '360',
+              viewer_count: 0,
+            },
+          ],
+        })
+    })
+
+    afterEach(function () {
+      return nock.cleanAll()
+    })
+
+    it('should return success response', async function () {
+      const result = await getStreams(
+        init({
+          api_key: 'eyJhbGciOiJ.eyJleHAi.B01hriveOMR',
+        })
+      )
+      // console.log(result)
+
+      expect(result.status.code).to.equal(200)
+    })
+
+    it('should return success response data', async function () {
+      const result = await getStreams(
+        init({
+          api_key: 'eyJhbGciOiJ.eyJleHAi.B01hriveOMR',
+        })
+      )
+
+      expect(result).to.be.an('object')
+      expect(result).to.have.property('data').to.be.an('array')
+      for (let p of result.data) {
```

Comment:
For something like an array, I think, you shouldn't loop the data. I think looping the data adds unnecessary logic to the test case. You can avoid the loop by using some built-in methods from chai. It's fine if you want to do deep equality comparison the object inside the array.

You can see these links for reference
https://medium.com/building-ibotta/testing-arrays-and-objects-with-chai-js-4b372310fe6d

https://codinglicks.com/blog/mocha-testing-cookbook-part-1/

https://stackoverflow.com/questions/17526805/chai-test-array-equality-doesnt-work-as-expected

Also one more thing, please refrain from using one character `p` for variable naming because it's hard to understand the meaning

Reply:
@faiq-naufal after I rethought about it I think using mas reference as per https://codinglicks.com/blog/mocha-testing-cookbook-part-1/#value-for-a-key actually enough to check the structure data because it will recheck as per a mock data right? Done at commit [f3e3db4](https://github.com/inlivedev/inlive-js-sdk/pull/13/commits/f3e3db4867b2de318e871a9319bd47eaa4769200)